Moggles/Controllers/ApplicationsController.cs

```diff
@@ -1,110 +1,79 @@
         {
-            _db = db;
+            _applicationsRepository = applicationsRepository;
         }
 
         [HttpGet]
-        public IActionResult GetAllApplications()
-
+        public async Task<IActionResult> GetAllApplications()
         {
-            return Ok(_db.Applications.ToList());
+            var allApps = await _applicationsRepository.GetAllAsync();
+            return Ok(allApps.OrderBy(a => a.AppName).ToList());
         }
 
         [HttpPost]
         [Route("add")]
-        public IActionResult AddApplication([FromBody]AddApplicationModel applicationModel)
+        public async Task<IActionResult> AddApplication([FromBody]AddApplicationModel applicationModel)
         {
             if (!ModelState.IsValid)
                 return BadRequest(ModelState);
 
-            var app = _db.Applications.FirstOrDefault(a => a.AppName == applicationModel.ApplicationName);
+            var apps = await _applicationsRepository.GetAllAsync();
+            var app = apps.FirstOrDefault(a => a.AppName == applicationModel.ApplicationName);
 
             if (app != null)
                 return BadRequest("Application already exists!");
-                
-            var application = _db.Applications.Add(new Application
-            {
-                AppName = applicationModel.ApplicationName
-            });
-
-            _db.SaveChanges();
-
-            _db.DeployEnvironments.Add(
-                new DeployEnvironment
-                {
-                    EnvName = applicationModel.EnvironmentName,
-                    ApplicationId = application.Entity.Id,
-                    DefaultToggleValue = applicationModel.DefaultToggleValue,
-                    SortOrder = 1
-                });
-
-            _db.SaveChanges();
-
-            return Ok(application.Entity);
+
+            var application = Application.Create(applicationModel.ApplicationName, applicationModel.EnvironmentName, applicationModel.DefaultToggleValue);
+
+            await _applicationsRepository.AddAsync(application);
+
+            return Ok(application);
         }
 
         [HttpPut]
         [Route("update")]
-        public IActionResult UpdateApplication([FromBody]UpdateApplicationModel applicationModel)
+        public async Task<IActionResult> UpdateApplication([FromBody]UpdateApplicationModel applicationModel)
         {
             if (!ModelState.IsValid)
                 return BadRequest(ModelState);
 
-            var app = _db.Applications.FirstOrDefault(a => a.Id == applicationModel.Id);
+            var app = await _applicationsRepository.FindByIdAsync(applicationModel.Id);
 
             if (app == null)
-                throw new InvalidOperationException("Application does not exists!");
-
-            app.AppName = applicationModel.ApplicationName;
+                return BadRequest("Application does not exist!");
 
-            _db.SaveChanges();
+            app.AppName = applicationModel.ApplicationName; 
```

Comment:
We should be using a method on the Domain object, like app.UpdateName(applicationModel.ApplicationName)